service/src/main/java/org/broadinstitute/listener/relay/inspectors/SamPermissionInspector.java

```diff
@@ -0,0 +1,72 @@
+package org.broadinstitute.listener.relay.inspectors;
+
+import com.microsoft.azure.relay.RelayedHttpListenerRequest;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import org.broadinstitute.listener.relay.inspectors.InspectorType.InspectorNameConstants;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Component;
+
+@Component(InspectorNameConstants.SAM_CHECKER)
+public class SamPermissionInspector implements RequestInspector {
+  private final Logger logger = LoggerFactory.getLogger(SamPermissionInspector.class);
+  private final SamResourceClient samResourceClient;
+
+  public SamPermissionInspector(SamResourceClient samResourceClient) {
+    this.samResourceClient = samResourceClient;
+  }
+
+  @Override
+  public boolean inspectWebSocketUpgradeRequest(
+      @NonNull RelayedHttpListenerRequest relayedHttpListenerRequest) {
+    return checkPermission(relayedHttpListenerRequest.getHeaders());
+  }
+
+  @Override
+  public boolean inspectRelayedHttpRequest(
+      @NonNull RelayedHttpListenerRequest relayedHttpListenerRequest) {
+    return checkPermission(relayedHttpListenerRequest.getHeaders());
+  }
+
+  private boolean checkPermission(Map<String, String> headers) {
+
+    if (headers == null) {
+      logger.error("No auth headers found");
+      return false;
+    }
+
+    var cookieValue = headers.getOrDefault("cookie", headers.get("Cookie"));
+
+    var leoToken = getToken(cookieValue);
+
+    if (leoToken.isEmpty()) {
+      logger.error("No valid cookie found");
+      return false;
+    } else {
+      var token = leoToken.get();
+      return checkCachedPermission(token);
+    }
+  }
+
+  public boolean checkCachedPermission(String accessToken) {
+    var expiresAt = samResourceClient.checkWritePermission(accessToken);
+    return expiresAt.isAfter(Instant.now());
+  }
+
+  protected Optional<String> getToken(@NotNull String cookieValue) {
+    String[] splitted = cookieValue.split(";");
+
+    if(cookieValue.isEmpty())
+      return Optional.empty();
+    else
+      return Arrays.stream(splitted)
+        .filter(s -> s.contains("LeoToken"))
+        .findFirst()
+        .map(s -> s.split("=")[1]);
```

Comment:
Google java code style (which all terra follows) requires brackets around all statements. So this needs some `{`s.

Although in this case the `if` is unnecessary. You can use `Arrays.stream(splitted)` even if `cookieValue.isEmpty()`.
```suggestion
      return Arrays.stream(splitted)
        .filter(s -> s.contains("LeoToken"))
        .findFirst()
        .map(s -> s.split("=")[1]);
```

Reply:
>Google java code style (which all terra follows) requires brackets around all statements.
dang...I don't wanna write ";" if java allows it